### PR TITLE
fix(@angular-devkit/build-angular): use istanbul-lib-instrument directly for karma code coverage

### PR DESCRIPTION
**Pull request by clydin**
The `istanbul-lib-instrument` package provides the required functionality needed to instrument code for test coverage within the context of the Angular CLI. Since the build pipeline already contains a customized babel preset, this package can be integrated directly into the pipeline.
This reduces the number of dependencies required for `@angular-devkit/build-angular` including the deprecated `inflight` package.

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/builders/karma/tests/behavior/code-coverage_spec.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ describeBuilder(execute, KARMA_BUILDER_INFO, (harness) => {
         .toPromise();
     });
 
-    it('should remapped instrumented code back to the original source', async () => {
+    fit('should remapped instrumented code back to the original source', async () => {
```
Comment (eneajaho) [inline packages/angular_devkit/build_angular/src/builders/karma/tests/behavior/code-coverage_spec.ts:89]:
```suggestion
    it('should remapped instrumented code back to the original source', async () => {
```

File: packages/angular_devkit/build_angular/src/tools/babel/plugins/add-code-coverage.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @returns A babel plugin object instance.
+ */
+export default function (): PluginObj {
+  const visitors = new WeakMap<NodePath, Visitor>();
+
+  return {
+    visitor: {
+      Program: {
+        enter(path, state) {
+          const visitor = programVisitor(types, state.filename, {
```
Comment (alan-agius4) [inline packages/angular_devkit/build_angular/src/tools/babel/plugins/add-code-coverage.ts:25]: - The exclusion of `options.instrumentCode.includedBasePath` leads to instrumenting files outside the project scope.
- To maintain the current functionality, `node_modules` should also be excluded from instrumentation.

Comment (clydin) [reply packages/angular_devkit/build_angular/src/tools/babel/plugins/add-code-coverage.ts:25]: That's handled here:
https://github.com/angular/angular-cli/blob/6c6985489f24ee820a9b08c9d09b78bd4ea18f3d/packages/angular_devkit/build_angular/src/tools/babel/webpack-loader.ts#L150-L155

**Review by alan-agius4 — APPROVED** [review]

**Comment by alan-agius4** [comment]
@clydin, this does not apply cleanly on the patch branch.

**Pull request merged**